src/core/geometry/qgsgeometry.cpp

```diff
@@ -233,6 +233,9 @@ QgsGeometry QgsGeometry::fromMultiPolygonXY( const QgsMultiPolygonXY &multipoly
 
 QgsGeometry QgsGeometry::fromRect( const QgsRectangle &rect )
 {
+  if ( rect.isNull() )
```

Comment:
Let's leave this for the other pr, and just get the function rename only bit merged first